src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -0,0 +1,899 @@
+    )
+    if 'omsagent' in addon_profiles:
+        _ensure_container_insights_for_monitoring(cmd, addon_profiles['omsagent'])
+    aad_profile = None
+    if any([aad_client_app_id, aad_server_app_id, aad_server_app_secret, aad_tenant_id]):
+        aad_profile = ManagedClusterAADProfile(
+            client_app_id=aad_client_app_id,
+            server_app_id=aad_server_app_id,
+            server_app_secret=aad_server_app_secret,
+            tenant_id=aad_tenant_id
+        )
+
+    # Check that both --disable-rbac and --enable-rbac weren't provided
+    if all([disable_rbac, enable_rbac]):
+        raise CLIError('specify either "--disable-rbac" or "--enable-rbac", not both.')
+
+    mc = ManagedCluster(
+        location=location, tags=tags,
+        dns_prefix=dns_name_prefix,
+        kubernetes_version=kubernetes_version,
+        enable_rbac=False if disable_rbac else True,
+        agent_pool_profiles=[agent_pool_profile],
+        linux_profile=linux_profile,
+        service_principal_profile=service_principal_profile,
+        network_profile=network_profile,
+        addon_profiles=addon_profiles,
+        aad_profile=aad_profile)
+
+    # Due to SPN replication latency, we do a few retries here
+    max_retry = 30
+    retry_exception = Exception(None)
+    for _ in range(0, max_retry):
+        try:
+            return sdk_no_wait(no_wait, client.managed_clusters.create_or_update,
+                               resource_group_name=resource_group_name, resource_name=name, parameters=mc)
+        except CloudError as ex:
+            retry_exception = ex
+            if 'not found in Active Directory tenant' in ex.message:
+                time.sleep(3)
+            else:
+                raise ex
+    raise retry_exception
+
+
+def aks_update(cmd, client, resource_group_name, name, enable_cluster_autoscaler=False,
+               disable_cluster_autoscaler=False,
+               update_cluster_autoscaler=False,
+               min_count=None, max_count=None, no_wait=False):
+    if (all([enable_cluster_autoscaler, disable_cluster_autoscaler]) or
+            all([enable_cluster_autoscaler, update_cluster_autoscaler]) or
+            all([disable_cluster_autoscaler, update_cluster_autoscaler])):
+        raise CLIError('specify either "--enable_cluster_autoscaler" or '
+                       '"--disable_cluster_autoscaler" or '
+                       '"--update_cluster_autoscaler".')
+    if all([not enable_cluster_autoscaler, not disable_cluster_autoscaler, not update_cluster_autoscaler]):
+        raise CLIError('specify either "--enable_cluster_autoscaler" or '
+                       '"--disable_cluster_autoscaler" or '
+                       '"--update_cluster_autoscaler".')
+
+    # TODO: change this approach when we support multiple agent pools.
+    instance = client.managed_clusters.get(resource_group_name, name)
+    node_count = instance.agent_pool_profiles[0].count
+
+    if not min_count or not max_count:
```

Comment:
this might not work:

if min_count is set to 0 (but I believe you check before that it has to be at least 1 so this might not apply), since 0 int would convert into false. anyways a better check would be

```if min_count is None or max_count is None:```

This applies to the if on line 505 too (but is not None... etc), my only worry would be if that min_count requirement changes to support 0 as min_count but not sure with the specific requirements or if this one day will be valid.

Reply:
There is an email thread talking about if node count can be 0, and the discussion is still on. I am not sure in the future, if 0 can be allowed or not. But your concern makes sense. I can change that.